core/src/main/kotlin/no/ntnu/beardblaster/screen/LobbyScreen.kt

```diff
@@ -129,29 +142,56 @@ class LobbyScreen(
-                                GameData.instance.game = it.data
-                                opponentLabel.setText("${it.data.opponent?.displayName} - ${it.data.opponent?.beardLength}cm")
-                                startGameBtn.isVisible = true
-                            } else {
-                                opponentLabel.setText("Waiting for opponent to join");
-                                startGameBtn.isVisible = false
+            LOG.debug { "Game created - Subscribing to new game." }
+            LOG.debug { "Should I subscribe? ${!::subscription.isInitialized || !subscription.isActive}." }
+            LOG.debug { "Is initialized? ${::subscription.isInitialized}." }
+            LOG.debug { "Is active? ${::subscription.isInitialized && subscription.isActive}." }
+
+            if(::subscription.isInitialized && subscription.isActive) {
+                subscription.cancel()
+            }
+
+
+            if(!::subscription.isInitialized || !subscription.isActive) {
+                // Yeah, this is ugly! Listen for live updates on lobby with id
+                subscription = KtxAsync.launch {
+                    LOG.debug { "Running subscribe to game with id ${p1.id}" }
+
+                    LobbyRepository().subscribeToLobbyUpdates(p1.id).collect {
+                        when (it) {
+                            is State.Success -> {
+                                // On received update: Check if opponent of updated Game is not null
+                                if (it.data.opponent != null) {
+                                    // Player may now start the game
+                                    GameData.instance.game = it.data
+                                    infoLabel.setText("A worthy opponent joined!")
+                                    opponentLabel.setText("${it.data.opponent?.displayName}")
+                                    opponentBeardLabel.setText("${it.data.opponent?.beardLength}cm")
+                                    opponentBeardLabel.color = BeardScale.getBeardColor(it.data.opponent?.beardLength?:0f)
+                                    startGameBtn.isVisible = true
+                                } else {
+                                    opponentLabel.setText("Waiting for opponent to join");
+                                    infoLabel.setText(Nls.shareGameCodeMessage())
+                                    startGameBtn.isVisible = false
+                                }
+                            }
+                            is State.Loading -> {
+                            }
+                            is State.Failed -> {
+                                LOG.error { it.message }
                             }
-                        }
-                        is State.Loading -> {
-                        }
-                        is State.Failed -> {
                         }
                     }
                 }
             }
         }
     }
+
```

Comment:
Just as a note, according to my research, the dispose method is not called automatically by the screens themselves (which seems strange to me, but ok).